Сделал задачу "Отправка донатов"

ringoway opened the pull request:
Написал тесты, тесты проходят. Через постман не проходит, так как не обнаруживает User'а.

ringoway committed:
Сделал задачу "Отправка донатов".

kkkooolllyyyaaa commented on src/main/java/faang/school/projectservice/service/DonationService.java:
```diff
@@ -0,0 +1,52 @@
+    private final UserServiceClient userServiceClient;
+
+    public DonationDto send(DonationDto donationDto) {
+        donationServiceValidator.isUserExist(userServiceClient, donationDto); /* по какой-то причине user не находится и кидается ошибка
+        "This user doesn't exist" в валидаторе */
```
Наверное потому что у тебя user service не запущен, или в БД не записан. Попробуй запустить и создать там юзера с конкретным айди и потестить через postman, иначе как ты поймешь, что всё работает)

kkkooolllyyyaaa commented on src/main/java/faang/school/projectservice/util/validator/DonationServiceValidator.java:
```diff
@@ -0,0 +1,29 @@
+import feign.FeignException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DonationServiceValidator {
+    public void validateStatus(Campaign campaign) {
+        if (campaign.getStatus() != CampaignStatus.ACTIVE) {
+            throw new EntityStatusException("Campaign is not active");
+        }
+    }
+
+    public void isUserExist(UserServiceClient userServiceClient, DonationDto donationDto) {
+        try {
+            userServiceClient.getUser(donationDto.getUserId());
+        } catch (FeignException.FeignClientException exception) {
+            throw new UserNotFoundException("This user doesn't exist");
+        }
+    }
+}
```
Давай это не выносить в отдельный класс, лучше сразу в методах сервиса это проверять.

kkkooolllyyyaaa reviewed (changes requested):
Надо потестить через постман нормально)

ringoway committed:
Исправил ошибки, протестировал через постман.
Ошибки заключались в:
 1. Необходимости запустить payment_service
 2. Исправить пути в UserServiceClient

kkkooolllyyyaaa reviewed (approved):
👍
